src/Architectures.jl

```diff
@@ -60,19 +60,31 @@ Abstract supertype for Distributed architectures supported by Oceananigans.
 abstract type AbstractMultiArchitecture <: AbstractArchitecture end
 
 """
-    CPU <: AbstractArchitecture
+    CPU()
 
-Run Oceananigans on one CPU node. Uses multiple threads if the environment
-variable `JULIA_NUM_THREADS` is set.
+Returns a CPU architecture for building Oceananigans models on a single CPU node.
 """
 struct CPU <: AbstractArchitecture end
 
+struct GPU{D} <: AbstractArchitecture
+    device :: D
+end
+
 """
-    GPU <: AbstractArchitecture
+    GPU(device)
+
+Returns a GPU architecture for building Oceananigans models on a GPU `device`.
 
-Run Oceananigans on a single NVIDIA CUDA GPU.
+Currently, only Nvidia CUDA GPUs are supported.
 """
-struct GPU <: AbstractArchitecture end
+function GPU()
+    if CUDA.has_cuda_gpu() 
+        return GPU(CUDA.device()) 
+    else 
+        @warn "No cuda capable device found! Reconsider running with architecture = CPU()" 
```

Comment:
@simone-silvestri isn't "cuda" capitalized? We capitalize it everywhere else, and the Julia package is "CUDA.jl". Let us know...